pytorch_transformers/tokenization_roberta.py

```diff
@@ -178,89 +179,62 @@ class RobertaTokenizer(PreTrainedTokenizer):
     RoBERTa tokenizer. Peculiarities:
         - GPT-2 tokenizer with a different integer mapping on top.
     """
-    vocab_files_names = VOCAB_FILES_NAMES
-    pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
+    vocab_files_names = DICT_FILES_NAMES
+    pretrained_vocab_files_map = PRETRAINED_DICT_FILES_MAP
     max_model_input_sizes = PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES
 
-    def __init__(self, vocab_file,
-                 bos_token="<s>", eos_token="</s>", **kwargs):
-        super(RobertaTokenizer, self).__init__(cls_token=bos_token, sep_token=eos_token, eos_token=eos_token, **kwargs)
+    def __init__(self, dict_file, bpe_tokenizer=None, bos_token="<s>", eos_token="</s>", sep_token="</s>", cls_token="<s>",
+                 unk_token="<unk>", **kwargs):
+        super(RobertaTokenizer, self).__init__(cls_token=bos_token, sep_token=eos_token, eos_token=eos_token,
+                                               unk_token=unk_token, **kwargs)
 
-        self.gpt2_tokenizer = GPT2Tokenizer.from_pretrained('gpt2')
-        self.dictionary = Dictionary.load(vocab_file)
+        self.gpt2_tokenizer = GPT2Tokenizer.from_pretrained("gpt2") if bpe_tokenizer is None else bpe_tokenizer
```

Comment:
I'm not a fan of calling this classmethod in `__init__` and having a GPT2Tokenizer as class attribute.

Do you think you could export a clean vocabulary for RobertA with the added tokens in the right position and inherit directly from GPT2Tokenizer, just loading the RobertA vocabulary instead of the GPT2 one?

This would let people reuse the tokenizer more easily as it would behave the same way than all the other tokenizers.

Or am I missing something there? I don't really understand why this tokenizer has to be so different from the others.

Reply:
I agree that changing the RoBERTa tokenizer to be fully independent instead of a wrapper would better respect the current library architecture and be easier to test/use. I'll work on that.